perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -1112,6 +1112,23 @@ public int updateFormItems(PerunSession sess, ApplicationForm form, List<Applica
 			// is item to create ? => create
 			if (item.getId() <= 0 && !item.isForDelete()) {
 				int temporaryId = item.getId();
+				// Check if newly created item contains allowed tags (in case of Header and HTML Comment
+				if (item.getType() == HTML_COMMENT || item.getType() == HEADING) {
```

Comment:
we could extract the repetitive logic from creating and updating to separate method to avoid code duplication, what do you think?

Reply:
The difference between these two blocks is that this one directly replaces the i18n of the item, whereas the following function does not. The main reason for it is there is a partial update in the following function (i.e. only ItemTexts is directly updated).

Reply:
Aha, I see. thanks for clarification